src/sources/soundsourceproxy.cpp

```diff
@@ -485,64 +485,71 @@ mixxx::AudioSourcePointer SoundSourceProxy::openAudioSource(const mixxx::AudioSo
-                              << "using mode"
-                              << openMode;
-            // Continue with the next SoundSource provider
-            nextSoundSourceProvider();
-            if (!getSoundSourceProvider() && (openMode == mixxx::SoundSource::OpenMode::Strict)) {
-                // No provider was able to open the source in Strict mode.
-                // Retry to open the file in Permissive mode starting with
-                // the first provider...
-                m_soundSourceProviderRegistrationIndex = 0;
-                openMode = mixxx::SoundSource::OpenMode::Permissive;
-            }
-            initSoundSource();
-            continue; // try again
-        }
-        if ((openResult == mixxx::SoundSource::OpenResult::Succeeded) && m_pSoundSource->verifyReadable()) {
-            m_pAudioSource = mixxx::AudioSourceTrackProxy::create(m_pTrack, m_pSoundSource);
-            DEBUG_ASSERT(m_pAudioSource);
-            if (m_pAudioSource->frameIndexRange().empty()) {
-                kLogger.warning() << "File is empty"
-                                  << getUrl().toString();
-            }
-            // Overwrite metadata with actual audio properties
-            if (m_pTrack) {
-                m_pTrack->updateAudioPropertiesFromStream(
-                        m_pAudioSource->getStreamInfo());
+        if (openResult == mixxx::SoundSource::OpenResult::Succeeded) {
+            if (m_pSoundSource->verifyReadable()) {
+                m_pAudioSource = mixxx::AudioSourceTrackProxy::create(m_pTrack, m_pSoundSource);
+                DEBUG_ASSERT(m_pAudioSource);
+                if (m_pAudioSource->frameIndexRange().empty()) {
+                    kLogger.warning()
+                            << "File is empty"
+                            << getUrl().toString();
+                }
+                // Overwrite metadata with actual audio properties
+                if (m_pTrack) {
+                    m_pTrack->updateAudioPropertiesFromStream(
+                            m_pAudioSource->getStreamInfo());
+                }
+                return m_pAudioSource;
             }
+            m_pSoundSource->close(); // cleanup
         } else {
-            kLogger.warning() << "Failed to open file"
-                              << getUrl().toString()
-                              << "with provider"
-                              << getSoundSourceProvider()->getName();
-            if (openResult == mixxx::SoundSource::OpenResult::Succeeded) {
-                m_pSoundSource->close(); // cleanup
+            kLogger.warning()
+                    << "Failed to open file"
+                    << getUrl().toString()
+                    << "with provider"
+                    << getSoundSourceProvider()->getName();
+            if (openMode == mixxx::SoundSource::OpenMode::Permissive &&
+                    openResult == mixxx::SoundSource::OpenResult::Failed) {
```

Comment:
I think we can improve readability by using 
} else if (openResult == mixxx::SoundSource::OpenResult::Failed) {
...
} else {
    DEBUG_ASSERT(openResult == mixxx::SoundSource::OpenResult::Abort)
...

https://github.com/mixxxdj/mixxx/blob/9d86357e736e19a5a6ec6d3af5107d492b5e10c1/src/sources/audiosource.h#L176

Should mention the second path